MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCollectionViewCell/FavouriteButton.swift

```diff
@@ -0,0 +1,23 @@
+import UIKit
+
+class FavouriteButton: UIButton {
+    
+    let size: CGFloat = 32
+        
+    init() {
+        let rect = CGRect(x: 0, y: 0, width: size, height: size)
+        super.init(frame: rect)
+        
+        setImage(ImageEnum.favouriteIcon.image, for: .normal)
+        setBackgroundImage(ImageEnum.favButtonBackground.image, for: .normal)
+        imageView?.snp.makeConstraints {
+            $0.center.equalToSuperview()
+            $0.height.equalTo(size / 2)
+        }
```

Comment:
We've talked that this should be in an extension which can be in this file.

Reply:
[Conformed](https://github.com/lihter/Movie-App/blob/100494de867d891c3e977dfcad60614d2b0b816d/MovieApp/MovieApp/App/Features/MovieBrowser/Components/MovieCollectionViewCell/FavouriteButton.swift#L20) FavouriteButton to DesignProtocol.